apps/src/studio/studio.js

```diff
@@ -2272,27 +2286,29 @@ Studio.reset = function (first) {
     if (Studio.sprite[i]) {
       Studio.sprite[i].removeElement();
     }
+    var spriteStart = Studio.spriteStart_[i];
     Studio.sprite[i] = new Sprite({
-      x: Studio.spriteStart_[i].x,
-      y: Studio.spriteStart_[i].y,
-      displayX: Studio.spriteStart_[i].x,
-      displayY: Studio.spriteStart_[i].y,
+      x: spriteStart.x,
+      y: spriteStart.y,
+      displayX: spriteStart.x,
+      displayY: spriteStart.y,
       loop: true,
-      speed: constants.DEFAULT_SPRITE_SPEED,
-      size: constants.DEFAULT_SPRITE_SIZE,
-      dir: Direction.NONE,
-      displayDir: Direction.SOUTH,
-      emotion: level.defaultEmotion || Emotions.NORMAL,
+      speed: spriteStart.speed || constants.DEFAULT_SPRITE_SPEED,
+      size: spriteStart.size || constants.DEFAULT_SPRITE_SIZE,
+      dir: spriteStart.direction || Direction.NONE,
+      displayDir: spriteStart.direction || Direction.SOUTH,
+      emotion: spriteStart.emotion || level.defaultEmotion || Emotions.NORMAL,
       renderOffset: renderOffset,
       // tickCount of last time sprite moved,
       lastMove: Infinity,
       // overridden as soon as we call setSprite
       visible: !level.spritesHiddenToStart
     });
 
+    var sprite = spriteStart.sprite || (i % Studio.startAvatars.length);
```

Comment:
I considered that, but the next line uses `spriteIndex` to refer to a slightly different concept of Sprite and Index.